internal/tools/aws/iam.go

```diff
@@ -140,16 +156,20 @@ func (a *Client) iamEnsureUserDeleted(awsID string, logger log.FieldLogger) erro
 	return nil
 }
 
-func (a *Client) iamEnsureS3PolicyCreated(awsID, policyARN, bucketName, permittedDirectory string, logger log.FieldLogger) (*iam.Policy, error) {
-	getResult, err := a.Service().iam.GetPolicy(&iam.GetPolicyInput{
-		PolicyArn: aws.String(policyARN),
-	})
+func (a *Client) iamEnsureS3PolicyCreated(awsID, policyARN, bucketName, permittedDirectory string, logger log.FieldLogger) (*types.Policy, error) {
+	ctx := context.TODO()
+	getResult, err := a.Service().iam.GetPolicy(
+		ctx,
+		&iam.GetPolicyInput{
+			PolicyArn: aws.String(policyARN),
+		})
 	if err == nil {
 		logger.WithField("iam-policy-name", *getResult.Policy.PolicyName).Debug("AWS IAM policy already created")
 		return getResult.Policy, nil
 	}
-	if aerr, ok := err.(awserr.Error); ok {
-		if aerr.Code() != iam.ErrCodeNoSuchEntityException {
+	if err != nil {
```

Comment:
We do not need this `if/else` check anymore

Reply:
Please review this simplified logic change: https://github.com/mattermost/mattermost-cloud/pull/803/commits/8e02ca327ea0a08b0d6a458b4b2e3c62a22014f2